src/interaction/inspector.jl

```diff
@@ -662,7 +662,7 @@ function show_imagelike(inspector, plot, name, edge_based)
     a._color[] = if z isa AbstractFloat
         interpolated_getindex(
             to_colormap(plot.colormap[]), z,
-            to_value(get(plot.attributes, :colorrange, (0, 1)))
+            get_attribute(plot.attributes, :colorrange, (0, 1))
```

Comment:
I guess? I'm not used to that yet, but it should probably be used in a lot of these functions

Reply:
Using `calculated_colors[].colorrange` instead does seem preferable as it contains the correct range so the actual values will be passed to `interpolated_getindex`.

Otherwise if the `colorrange` attribute is unset it will fallback to the default and the values will be clamped between 0 and 1 so if the data has values outside of this interval then `a._color[]` will only vary between the two endpoint colors which can easily be seen on the indicator for heatmaps and images with `interpolate = true`; using `calculated_colors` fixes this.

Shall I change that line to either `to_value(plot.attributes.calculated_colors[].colorrange)` or `plot.attributes.calculated_colors[].colorrange[]`, or is there a preferred method?

Reply:
Either is fine. `to_value` is generally a bit safer because it doesn't error on values (as opposed to observables), but `calculated_colors` is strictly typed to contain observables.

Reply:
We should have a function `extract_colorrange(plot)` that does this more generically.
Could work similarly like this: https://github.com/MakieOrg/Makie.jl/blob/master/src/makielayout/blocks/colorbar.jl#L28-L44
But just returning colorrange instead of constructing the ColorMapping etc...

Reply:
Something like this?
```diff --git a/src/makielayout/blocks/colorbar.jl b/src/makielayout/blocks/colorbar.jl
index 75b1a34a..3b8e9d2a 100644
--- a/src/makielayout/blocks/colorbar.jl
+++ b/src/makielayout/blocks/colorbar.jl
@@ -25,6 +25,16 @@ function colorbar_check(keys, kwargs_keys)
     end
 end
 
+function extract_colorrange(@nospecialize(plot::AbstractPlot))
+    if haskey(plot, :calculated_colors) && plot.calculated_colors[] isa Makie.ColorMapping
+        return plot.calculated_colors[].colorrange[]
+    elseif haskey(plot, :colorrange) && !(plot.colorrange[] isa Makie.Automatic)
+        return plot.colorrange[]
+    else
+        return (0, 1)
+    end
+end
+
 function extract_colormap(@nospecialize(plot::AbstractPlot))
     has_colorrange = haskey(plot, :colorrange) && !(plot.colorrange[] isa Makie.Automatic)
     if haskey(plot, :calculated_colors) && plot.calculated_colors[] isa Makie.ColorMapping

Reply:
Yes, that looks good!

Reply:
we can also use `Vec` if that's easier, I don't really mind. Any preferences @SimonDanisch ?

Reply:
Yeah lets use Vec if that already gets returned! Makes the conversion easier as well.